exporter/opentelemetry-exporter-jaeger/src/opentelemetry/exporter/jaeger/__init__.py

```diff
@@ -109,23 +109,26 @@ def __init__(
         service_name,
         agent_host_name=DEFAULT_AGENT_HOST_NAME,
         agent_port=DEFAULT_AGENT_PORT,
-        collector_host_name=None,
-        collector_port=None,
-        collector_endpoint=DEFAULT_COLLECTOR_ENDPOINT,
-        collector_protocol=DEFAULT_COLLECTOR_PROTOCOL,
+        collector_endpoint=None,
         username=None,
         password=None,
     ):
         self.service_name = service_name
-        self.agent_host_name = agent_host_name
-        self.agent_port = agent_port
+        self.agent_host_name = (
+            os.environ.get(OTEL_ENVS["agent_host"]) or agent_host_name
+        )
+        self.agent_port = (
+            int(os.environ.get(OTEL_ENVS["agent_port"]))
+            if os.environ.get(OTEL_ENVS["agent_port"])
+            else agent_port
+        )
         self._agent_client = None
-        self.collector_host_name = collector_host_name
-        self.collector_port = collector_port
-        self.collector_endpoint = collector_endpoint
-        self.collector_protocol = collector_protocol
-        self.username = username
-        self.password = password
+        self.collector_endpoint = (
+            os.environ.get(OTEL_ENVS["collector_endpoint"])
+            or collector_endpoint
+        )
+        self.username = os.environ.get(OTEL_ENVS["username"]) or username
```

Comment:
I would expect the or to be reversed here `username or os.environ.get(OTEL_ENVS["username"])` as to ensure that if someone overrides the variable in the code, it takes precedence over environment variables. It's the model I followed for the zipkin environment variables https://github.com/open-telemetry/opentelemetry-python/pull/1064/files#diff-eb584b1fd47b0e9ede2d87c686b87a1aR113. We should be consistent one way or another.

Reply:
I've added requested changes :)